reactive/kubernetes_master.py

```diff
@@ -2402,3 +2408,18 @@ def get_dns_provider():
 
     leader_set(auto_dns_provider=dns_provider)
     return dns_provider
+
+
+@when('kube-control.connected')
```

Comment:
@hyperbolic2346 one thing i hadn't considered until now... Should we also gate `send_registry_location` with an `is_leader` flag?

I'm not sure the ramifications of multiple masters sending registry locations to the workers, but I imagine it'll at least mean the workers/masters exchange unnecessary chatter.  Seems like we'd be better served if only the lead master send the registry config.

Reply:
@kwmonroe unnecessary chatter is not good, but would we need to be concerned with units that used to be leaders still having data set on the relation?

Reply:
Ah, right.. without relation-setting (or un-setting, in this case), a leader could be configured, send data, then be reconfigured, lose leadership, then have workers asking about that data from a non-leader -- without an anti handler, workers would continue getting invalid config. At least with all masters talking about the same config, all workers will get it right.